main.nf

```diff
@@ -877,12 +936,12 @@ if (merge) {
   }
 
 } else {
-  maps_to_merge.tap {
+  maps_to_merge_for_quantification.tap {
     maps_to_quantify
   }.map {
     [it[0], it[2], it[3]]
   }.tap {
-    maps_to_assemble
+    maps_to_merge_for_assembly
```

Comment:
Something wrong is going on here. The current merge process should be duplicated into `merge_for_assembly` and `merge_for_quantification`. Each process will get its own if-else case. Maps merged for quantification and maps merged for assembly should not be in contact.